### ISSUE TITLE
Tokenizer trait and auto-detection from GGUF

### ISSUE THREAD

**Issue by anibjoshi** [strata-ai-labs/strata-inference#6]
### Context
Different models use different tokenizers. The GGUF `tokenizer.ggml.model` key tells us which one:
- `"gpt2"` → BPE (LLaMA, Gemma, GPT-2)
- `"llama"` → SentencePiece/SPM
- `"bert"` → WordPiece (BERT, MiniLM)

We need a trait that abstracts over these and auto-selects based on GGUF metadata.

### Design
```rust
pub trait Tokenizer: Send + Sync {
    fn encode(&self, text: &str, add_special: bool) -> Vec<u32>;
    fn decode(&self, ids: &[u32]) -> String;
    fn vocab_size(&self) -> usize;
    fn bos_token_id(&self) -> Option<u32>;
    fn eos_token_id(&self) -> Option<u32>;
    fn pad_token_id(&self) -> Option<u32>;
}

impl Tokenizer {
    /// Auto-detect and create tokenizer from GGUF metadata
    pub fn from_gguf(gguf: &GgufFile) -> Result<Box<dyn Tokenizer>>;
}
```

### Acceptance Criteria
- [ ] `Tokenizer` trait with encode/decode/special tokens
- [ ] `from_gguf()` factory that reads `tokenizer.ggml.model` and dispatches
- [ ] Batch encoding: `fn encode_batch(texts: &[&str]) -> Vec<Vec<u32>>`
- [ ] Padding to max length within batch
- [ ] Attention mask generation: `fn attention_mask(ids: &[Vec<u32>]) -> Vec<Vec<u8>>`
- [ ] Test with both BPE (Gemma) and WordPiece (BERT) GGUF files

### Estimated LOC: ~200-300
